packages/twenty-front/src/modules/workflow/hooks/useRightDrawerWorkflowSelectAction.tsx

```diff
@@ -0,0 +1,129 @@
+import { useTabList } from '@/ui/layout/tab/hooks/useTabList';
+import { useCreateNode } from '@/workflow/hooks/useCreateNode';
+import { createStepFromParentStepIdState } from '@/workflow/states/createStepFromParentStepIdState';
+import { showPageWorkflowDiagramTriggerNodeSelectionState } from '@/workflow/states/showPageWorkflowDiagramTriggerNodeSelectionState';
+import { Workflow } from '@/workflow/types/Workflow';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import {
+  IconPlaystationSquare,
+  IconPlug,
+  IconPlus,
+  IconSearch,
+  IconSettingsAutomation,
+} from 'twenty-ui';
+import { v4 } from 'uuid';
+import { logError } from '~/utils/logError';
+
+export const useRightDrawerWorkflowSelectAction = ({
```

Comment:
1. on the project, we don't create "umbrella" hooks. We want to create one hook per specific purpose
Here we would create useRightDrawerWorkflowSelectActionTab, useRightDrawerWorkflowSelectActionOptions, ...

2. In your design you should distinguish business concerns (action.isCustom) from UI concerns (tab). Options should depend on action.isCustom and not from selected tab.



However, here I feel this could simplified:
- extract tabs into a constant, no need to have it in a hook
- only introduce useComputeWorkflowSelectActionOption(action)

3. I don't see tab in figmas

Reply:
In my next PR, the scope of the hook is restricted. I will make sure to rename the hook according to your recommendation.

See https://github.com/twentyhq/twenty/blob/b676de84abc13ee0ae0199d0d07c73cbe0e4051c/packages/twenty-front/src/modules/workflow/hooks/useRightDrawerWorkflowSelectAction.tsx#L13

Reply:
Noted for the distinction I should make about the different concerns in the hook. In my next PR I dropped tabs (which were in fact an old version in the Figma) so this is no longer a proble. I will keep in mind in the future.

Reply:
@Devessier please also rename this hook in the next PR so it details the actual logic

Reply:
I will do in my next PR as I already rewrote this part.